bin/check-dns.rb

```diff
@@ -132,6 +133,19 @@ def check_against_regex(entries, regex)
     critical "Resolved #{config[:domain]} #{config[:type]} did not match #{regex}"
   end
 
+  def check_ips(entries)
+    ips = []
+    entries.answer.rrsets(config[:type]).each do |ip|
+      ips << ip.to_s.split("\t").last.tr("\n", '')
```

Comment:
Instead of turning the DNSRuby objects into strings and then parsing the strings, you can retrieve the IP addresses via `.rrs.map { |rr| rr.address.to_s }`

Or replace the `.each` with:

    ips = entries.answer.rrsets(config[:type]).flat_map(&:rrs).map(&:address).map(&:to_s)